src/connection.js

```diff
@@ -245,6 +245,11 @@ class Connection extends EventEmitter {
       } else {
         this.emit('rerouting');
       }
+      if (this.request) {
+        const err = RequestError('Connection closed before request completed.', 'ECLOSE');
```

Comment:
Thanks for merging the change. Looks like we're using the non-new form of RequestError everywhere, so leaving as is.